### PR TITLE
alicloud_ess_scalingconfiguration: Adds new attributes spot_strategy and spot_price_limit

### PR DESCRIPTION
**Pull request by TalentNing**
=== RUN   TestAccAlicloudEssScalingconfigurationsDataSource
--- PASS: TestAccAlicloudEssScalingconfigurationsDataSource (81.92s)
PASS
ok      github.com/aliyun/terraform-provider-alicloud/alicloud  83.627s

=== RUN   TestAccAlicloudEssScalingconfigurationsDataSource
--- PASS: TestAccAlicloudEssScalingconfigurationsDataSource (83.38s)
PASS
ok      github.com/aliyun/terraform-provider-alicloud/alicloud  85.091s

### CODE REVIEW

File: alicloud/resource_alicloud_ess_scaling_configuration_test.go [hunk 1]
```diff
@@ -430,6 +430,24 @@ func TestAccAlicloudEssScalingConfigurationPerformanceLevel(t *testing.T) {
...
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheck(map[string]string{
+						"spot_strategy":                    "SpotWithPriceLimit",
+						"spot_price_limit.#":               "1",
+						"spot_price_limit.0.instance_type": CHECKSET,
+						"spot_price_limit.0.price_limit":   "2.1",
+					}),
+				),
+			},
```
Comment (Wanghx0991) [inline alicloud/resource_alicloud_ess_scaling_configuration_test.go:450]: 需要增加一个Update的测试步

Comment (TalentNing) [reply alicloud/resource_alicloud_ess_scaling_configuration_test.go:461]: done

File: alicloud/resource_alicloud_ess_scaling_configuration.go [hunk 2]
```diff
@@ -291,6 +291,26 @@ func resourceAlicloudEssScalingConfiguration() *schema.Resource {
...
 				Optional: true,
 			},
+			"spot_strategy": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"spot_price_limit": {
+				Optional: true,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_ess_scaling_configuration.go:301]: 在resource中应该改为 TypeSet

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: alicloud/resource_alicloud_ess_scaling_configuration.go (hunk 2 above)
Comment (TalentNing) [reply alicloud/resource_alicloud_ess_scaling_configuration.go:301]: done

File: alicloud/data_source_alicloud_ess_scalingconfigurations.go [hunk 3]
```diff
@@ -147,6 +147,26 @@ func dataSourceAlicloudEssScalingConfigurations() *schema.Resource {
 							Type:     schema.TypeString,
 							Computed: true,
 						},
+						"spot_strategy": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"spot_price_limit": {
+							Type: schema.TypeSet,
+							Elem: &schema.Resource{
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_ess_scalingconfigurations.go:156]: 在datesource 中应该是 TypeList，在resource中是 TypeSet

**Pull request merged**